### PR TITLE
popup: doesn't invalidate zIndex when anchor set while popup is hidden (close #1214)

### LINKED ISSUE #1214: popup: doesn't invalidate zIndex when anchor set while popup is hidden
it was mine mistake when i worked on https://github.com/bem/bem-components/pull/972

### PR DESCRIPTION
**Pull request by persidskiy**
close #1214

### CODE REVIEW

**Comment by persidskiy** [comment]
@narqo look at this pls

File: common.blocks/popup/_target/popup_target_anchor.spec.js [hunk 1]
```diff
@@ -268,15 +264,18 @@ describe('popup', function() {
                     });
 
             popup.setMod('visible');
-            popup2.setAnchor(popupAnchorDomElem).setMod('visible');
 
-            Number(popup.domElem.css('z-index'))
-                .should.be.lt(Number(popup2.domElem.css('z-index')));
+            popup2.setAnchor(popupAnchorDomElem).setMod('visible');
+            getZIndex(popup).should.be.lt(getZIndex(popup2), 'initial');
```
Comment (narqo) [inline common.blocks/popup/_target/popup_target_anchor.spec.js]: Can you write more informative message in case the assertion fail?

Comment (narqo) [inline common.blocks/popup/_target/popup_target_anchor.spec.js]: I'm not sure, but I thought that  assertion `message` should contain addition information about particular error, e.g.

``` js
  › node -p "require('assert').fail(1, 0, 'one does not equal to zero')"

assert.js:92
  throw new assert.AssertionError({
        ^
AssertionError: one does not equal to zero
    at [eval]:1:19
```

From this point of view your messages in assertions look strange, sorry :)

File: common.blocks/popup/_target/popup_target_anchor.spec.js [hunk 2]
```diff
@@ -300,4 +299,8 @@ function buildPopup(parentDomElem, bemjson) {
         .bem('popup');
 }
 
+function getZIndex(popup) {
```
Comment (narqo) [inline common.blocks/popup/_target/popup_target_anchor.spec.js:322]: May be `getCssZIndex`? IMHO, this name looks more clear.

File: common.blocks/popup/_target/popup_target_anchor.spec.js [hunk 3]
```diff
@@ -268,15 +264,18 @@ describe('popup', function() {
...
             popup.setAnchor(popup2.domElem.find('span:first'));
+            getZIndex(popup).should.be.gt(getZIndex(popup2), 'popup moved to popup2 while visible');
 
-            Number(popup.domElem.css('z-index'))
-                .should.be.gt(Number(popup2.domElem.css('z-index')));
+            popup
+                .delMod('visible')
+                .setAnchor(popupAnchorDomElem)
+                .setMod('visible');
+            getZIndex(popup).should.be.lt(getZIndex(popup2), 'popup moved from popup2 while hidden');
```
Comment (narqo) [inline common.blocks/popup/_target/popup_target_anchor.spec.js]: May be we should move it to a separate test-suite? I think "popup should recalculate zIndexGroupLevel if anchor is changed" looks like a self sufficient statement in BDD terms

File: common.blocks/popup/_target/popup_target_anchor.spec.js (hunk 2 above)
Comment (persidskiy) [inline common.blocks/popup/_target/popup_target_anchor.spec.js:322]: no difference)
Is there any z-indexes not in css?

Comment (narqo) [inline common.blocks/popup/_target/popup_target_anchor.spec.js:322]: ZIndexGroupLevel / z-index-group_level_N

Comment (persidskiy) [inline common.blocks/popup/_target/popup_target_anchor.spec.js:322]: it is z-index-group, not z-index

**Comment by persidskiy** [comment]
:up:

File: common.blocks/popup/_target/popup_target_anchor.spec.js (hunk 2 above)
Comment (narqo) [inline common.blocks/popup/_target/popup_target_anchor.spec.js:322]: :(

**Pull request merged**